### PR TITLE
feat: Integrates an automatic update of the IRIS client.

### PR DESCRIPTION
**Pull request by jekutzsche**
* Integrates Watchtower to observe and update container versions.
* The Docker-Compose files use minor versions (1.0) rather than full
ones (1.0.1) now, so that they include all bugfix releases and can be
automatically updated across bugfixes.
* SemVer tags are now created for each image in the release (1.0.1, 1.0,
1 and latest).

Refs  iris-connect/iris-backlog#104

### CODE REVIEW

**Comment by hey-johnnypark** [comment]
Hi Jens,

Is there a possibility to deactivate watchtower, just for admins that would not like it. 

Andi

**Comment by jekutzsche** [comment]
Hi Andi @hey-johnnypark 

with the configuration `WATCHTOWER_MONITOR_ONLY: true` it is defacto deactivated. It should at the moment only check if there are new versions and log this. With the configurations of https://containrrr.dev/watchtower/, notifications and automatic updates can then still be configured.

File: infrastructure/deployment/docker-compose-ext-postgres.yml [hunk 1]
```diff
@@ -109,5 +109,18 @@ services:
       - iris-client
       - iris-frontend
 
+  # Watchtower for auto-updates
+  watchtower:
+    image: containrrr/watchtower
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (hey-johnnypark) [inline infrastructure/deployment/docker-compose-ext-postgres.yml:116]: As discussed. Let us check for OS other than X OSs

Comment (jekutzsche) [reply infrastructure/deployment/docker-compose-ext-postgres.yml:116]: I think I found a configuration for Windows and added it as a comment.

**Comment by jekutzsche** [comment]
> Hi Jens, Is there a possibility to deactivate watchtower, just for admins that would not like it. Andi

I have replaced `WATCHTOWER_MONITOR_ONLY: true` with `WATCHTOWER_NO_PULL: true`. So the images are no longer regularly pulled and there is no change to the system at the moment.

**Review by tboegner — APPROVED** [review]
Containers start, looks fine to me.

**Pull request merged**